app/services/billable_metrics/destroy_service.rb

```diff
@@ -26,12 +29,34 @@ def call
       # NOTE: Discard all related events asynchronously.
       BillableMetrics::DeleteEventsJob.perform_later(metric)
 
+      track_billable_metric_deleted
+
+      # NOTE: Refresh all invoices linked to the billable metric.
+      Invoice.find(draft_invoice_ids).each do |invoice|
+        ::Invoices::RefreshDraftService.call(invoice:)
```

Comment:
Shouldn't we execute it in job? We might have a lot of draft invoice with the metric and so it could take a while to refresh all of them. What do you think?

Reply:
Yeah - I agree - Good point 👍